ostd/src/mm/io.rs

```diff
@@ -387,6 +389,7 @@ pub trait FallibleVmWrite<F> {
 /// and physical address level. There is not guarantee for the operation results
 /// of `VmReader` and `VmWriter` in overlapping untyped addresses, and it is
 /// the user's responsibility to handle this situation.
+#[derive(Clone)]
```

Comment:
Manually implement the `Clone` to avoid implementing `Clone` for the marker types?
```rust
/// `Clone` can be implemented for `VmReader` because it points to either untyped memory or
/// immutable references. Note that we cannot implement `Clone` for `VmWriter` because it can point
/// to mutable references, which must be exclusive.
impl Clone for VmReader { ... }
```